### PR TITLE
refactor: use lifecycle on start process

### PR DESCRIPTION
**Pull request by fengmk2**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -2,6 +2,12 @@
...
 const oss = require('./lib/oss');
 
-module.exports = app => {
-  oss(app);
+module.exports = class {
+  constructor(app) {
+    this.app = app;
+  }
+
+  configDidLoad() {
```
Comment (fengmk2) [inline app.js]: 允许修改 oss 配置的机会。

File: app.js [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 const oss = require('./lib/oss');
 
-module.exports = app => {
-  oss(app);
+module.exports = class {
```
[the remark this answers is not in the document]
Comment (fengmk2) [reply app.js]: 堆栈足够清晰，没必要吧

[the remark this answers is not in the document]
Comment (atian25) [reply app.js]: ok

**Comment by fengmk2** [comment]
算 break 么？

**Pull request merged**